cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/SubgraphWalker.kt

```diff
@@ -55,7 +54,7 @@ object SubgraphWalker {
      * @param classType the class type
      * @return its fields, including the ones from its superclass
      */
-    private fun getAllFields(classType: Class<*>): Collection<Field> {
+    fun getAllFields(classType: Class<*>): Collection<Field> {
```

Comment:
In theory yes. Unfortunately the edge walker is slower in comparison to this very efficient implementation of only following the AST nodes forward (in the `testWalkerSpeed` the difference is about 200ms vs. 400ms). Because we are using the AST traversal quite a lot, I would like to keep this specialised version for AST nodes.